custom/icds/models.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import uuid
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.app_manager.dbaccessors import get_build_by_version
+from custom.icds.const import (
+    FILE_TYPE_CHOICE_ZIP,
+    FILE_TYPE_CHOICE_DOC,
+    DISPLAY_CHOICE_LIST,
+    DISPLAY_CHOICE_FOOTER,
+)
+from custom.icds.utils.ccz_hosting import CCZHostingUtility
+from custom.icds.validators import (
+    LowercaseAlphanumericValidator,
+)
+from custom.nic_compliance.utils import hash_password
+
+
+class CCZHostingLink(models.Model):
+    identifier = models.CharField(null=False, unique=True, max_length=255, db_index=True,
+                                  validators=[LowercaseAlphanumericValidator])
+    username = models.CharField(null=False, max_length=255)
+    # b64_aes_encrypt'ed raw password is stored in DB
+    password = models.CharField(null=False, max_length=255)
+    domain = models.CharField(null=False, max_length=255)
+    page_title = models.CharField(blank=True, max_length=255)
+
+    def to_json(self):
+        from custom.icds.serializers import CCZHostingLinkSerializer
+        return CCZHostingLinkSerializer(self).data
+
+    def save(self, *args, **kwargs):
+        if not self.pk:
+            self.password = hash_password(self.password)
+        self.full_clean()
+        super(CCZHostingLink, self).save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        for ccz_hosting in CCZHosting.objects.filter(link=self):
+            ccz_hosting.delete()
```

Comment:
is this not handled by `on_delete=CASCADE` on the `CCZHosting` foreign key?

Reply:
that would not call the delete method on CCZHosting but just delete the object.
I updated to use signal instead of that its not confusing
https://github.com/dimagi/commcare-hq/pull/24129/commits/2b905b38c2199971ad033d106199cbbd4088327e